network/src/commonMain/kotlin/dev/icerock/moko/network/features/LanguageFeature.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.network.features
+
+import io.ktor.client.HttpClient
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.features.HttpClientFeature
+import io.ktor.client.request.HttpRequestPipeline
+import io.ktor.client.request.header
+import io.ktor.util.AttributeKey
+
+class LanguageFeature private constructor(
+    private val mLanguageHeaderName: String,
+    private val mLanguageProvider: LanguageFeature.LanguageCodeProvider
+) {
+    class Config {
+        var languageHeaderName: String? = null
+        var languageCodeProvider: LanguageCodeProvider? = null
+        fun build() = LanguageFeature(
+            languageHeaderName ?: throw IllegalArgumentException("HeaderName should be contain"),
+            languageCodeProvider ?: throw IllegalArgumentException("LanguageCodeProvider should be contain")
+        )
+    }
+
+    companion object Feature : HttpClientFeature<Config, LanguageFeature> {
+        override val key = AttributeKey<LanguageFeature>("LanguageFeature")
+
+        override fun prepare(block: Config.() -> Unit) = Config().apply(block).build()
+
+        override fun install(feature: LanguageFeature, scope: HttpClient) {
+            scope.requestPipeline.intercept(HttpRequestPipeline.State) {
+                feature.mLanguageProvider.getLanguageCode()?.apply {
+                    context.header(feature.mLanguageHeaderName, this)
+                }
+            }
+        }
+    }
+
+    interface LanguageCodeProvider {
+        fun getLanguageCode(): String?
+    }
+}
+
+fun HttpClientConfig<*>.install(
+    languageHeaderName: String,
+    languageCodeProvider: LanguageFeature.LanguageCodeProvider
+) {
+    install(LanguageFeature) {
+        this.languageHeaderName = languageHeaderName
+        this.languageCodeProvider = languageCodeProvider
+    }
+}
```

Comment:
it's bad readable. `install` function is common and see two args version which feature-specific is not understandable

Reply:
Changed for this feature and for previously created